Engine/AlgorithmManager.cs

```diff
@@ -95,37 +85,33 @@ public TimeSpan CurrentTimeStepElapsed
         /// </summary>
         public readonly Func<IsolatorLimitResult> TimeLoopWithinLimits;
 
-        private readonly bool _liveMode;
-
         /// <summary>
         /// Quit state flag for the running algorithm. When true the user has requested the backtest stops through a Quit() method.
         /// </summary>
         /// <seealso cref="QCAlgorithm.Quit(String)"/>
-        public bool QuitState
-        {
-            get { return State == AlgorithmStatus.Deleted; }
-        }
+        public bool QuitState => State == AlgorithmStatus.Deleted;
 
         /// <summary>
         /// Gets the number of data points processed per second
         /// </summary>
-        public long DataPoints
-        {
-            get { return _dataPointCount; }
-        }
+        public long DataPoints { get; private set; }
 
         /// <summary>
         /// Initializes a new instance of the <see cref="AlgorithmManager"/> class
         /// </summary>
         /// <param name="liveMode">True if we're running in live mode, false for backtest mode</param>
         public AlgorithmManager(bool liveMode)
         {
+            _lock = new object();
+            var timeLoopMaximum
+                = TimeSpan.FromMinutes(Config.GetDouble("algorithm-manager-time-loop-maximum", 20));
```

Comment:
Perhaps outside the scope of this PR, but I often find it nice to remove `Config` access from types. In this case, the object instantiating the `AlgorithmManager` could simply pass in the value of  `algorithm-manager-time-loop-maximum`.

I think this is fine as is, simply sharing some of my opinionated styling :)